File: AllSamples/Samples/Samples_Source/indigo/remoteobjects/remoteobjects.hellonewremoteobject_cs/server/remotableobject.cs
using System; 

namespace Microsoft.Samples.MessageBus.Quickstarts.RemoteObjects
{
    public class RemotableObject : MarshalByRefObject
    {
        public RemotableObject()
        {
            Console.WriteLine("Object {0} has been created.", this.GetHashCode().ToString());
        }

        public string Hello(string name)
        {
            return String.Format("Hello, {0}. This is RemotableObject instance \"{1}\"", name, this.GetHashCode().ToString());
        }
    
        ~RemotableObject()
        {
            Console.WriteLine("Object {0} is being torn down.", this.GetHashCode().ToString());
        }
    }
}

File: AllSamples/Samples/Samples_Source/indigo/remoteobjects/remoteobjects.hellonewremoteobject_vb/server/remotableobject.vb
Imports System
Namespace Microsoft.Samples.MessageBus.Quickstarts.RemoteObjects
    Public Class RemotableObject
        Inherits MarshalByRefObject

        Public Sub New()
            Console.WriteLine("Object {0} has been created.", Me.GetHashCode().ToString())
        End Sub
        Public Function Hello(ByVal name As String) As String
            Return String.Format("Hello, {0}. This is RemotableObject instance ""{1}""", name, Me.GetHashCode().ToString())
        End Function
        Protected Overrides Sub Finalize()
            Console.WriteLine("Object {0} is being torn down.", Me.GetHashCode().ToString())
        End Sub
    End Class
End Namespace